app/ui-utils/client/lib/renderMessageBody.js

```diff
@@ -32,4 +33,4 @@ export const renderMessageBody = mem((message) => {
 
 	return (Array.isArray(tokens) ? tokens.reverse() : [])
 		.reduce((html, { token, text }) => html.replace(token, () => text), html);
-}, 500, ({ _id, _updatedAt }) => _id && _updatedAt && _id + _updatedAt);
+}, 500, (message) => md5(JSON.stringify(message)));
```

Comment:
Does this make an editing of the message re-render it? Or what's the motivation?

Reply:
Yes, editing will re-renders, but editing would also change `_updatedAt`, so that the message was re-rendered with the previous code as well. The new code also re-renders if the translation changes.

However, instead of using `md5` library, I would prefer to just use `crypto`, which is already included in Node.js and also has the option to generate a md5 hash.

Reply:
Sth. like

    crypto.createHash('md5').update(message).digest("hex")

Reply:
```javascript

const createHashKey = function(o){
    return crypto.createHash('md5').update(String(JSON.stringify(o))).digest("hex")
}
```
and then use only the properties relevant to caching
```javascript
({_id, updatedAt, translations}) => createHashKey(...arguments)
```